File: sdk/src/main/java/com/cxense/cxensesdk/LoadCallback.java
package com.cxense.cxensesdk;

/**
 * Interface that is implemented to discover when data loading has finished.
 *
 * @param <T> success data type
 * @author Dmitriy Konopelkin (dmitry.konopelkin@cxense.com) on (2017-06-05).
 */
public interface LoadCallback<T> {
    /**
     * Called when load has completed successfully.
     *
     * @param data result data
     */
    void onSuccess(T data);

    /**
     * Called when load has completed with error.
     *
     * @param throwable error
     */
    void onError(Throwable throwable);
}

File: sdk/src/main/java/io/piano/android/cxense/LoadCallback.kt
package io.piano.android.cxense

/**
 * Interface that is implemented to discover when data loading has finished.
 *
 * @param <T> success data type
 */
public interface LoadCallback<in T : Any> {
    /**
     * Called when load has completed successfully.
     *
     * @param data result data
     */
    public fun onSuccess(data: T)

    /**
     * Called when load has completed with error.
     *
     * @param throwable error
     */
    public fun onError(throwable: Throwable)
}
